src/Schema/Directives/DropArgsDirective.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+namespace Nuwave\Lighthouse\Schema\Directives;
+
+use Closure;
+use GraphQL\Type\Definition\ResolveInfo;
+use Nuwave\Lighthouse\Execution\Arguments\ArgumentSet;
+use Nuwave\Lighthouse\Schema\Values\FieldValue;
+use Nuwave\Lighthouse\Support\Contracts\Directive;
+use Nuwave\Lighthouse\Support\Contracts\FieldMiddleware;
+use Nuwave\Lighthouse\Support\Contracts\GraphQLContext;
+use Nuwave\Lighthouse\Support\Utils;
+
+class DropArgsDirective extends BaseDirective implements FieldMiddleware
+{
+    public static function definition(): string
+    {
+        return /** @lang GraphQL */ <<<'GRAPHQL'
+"""
+Apply the @drop directives on the incoming arguments.
+"""
+directive @dropArgs on FIELD_DEFINITION
+GRAPHQL;
+    }
+
+    /**
+     * @param FieldValue $fieldValue
+     * @param Closure $next
+     * @return mixed
+     */
+    public function handleField(FieldValue $fieldValue, Closure $next)
+    {
+        $resolver = $fieldValue->getResolver();
+
+        return $next(
+            $fieldValue->setResolver(
+                function ($root, array $args, GraphQLContext $context, ResolveInfo $resolveInfo) use ($resolver) {
+                    $argumentSet = $resolveInfo->argumentSet;
+                    $this->drop($argumentSet);
+
+                    return $resolver(
+                        $root,
+                        $argumentSet->toArray(),
+                        $context,
+                        $resolveInfo
+                    );
+                }
+            )
+        );
+    }
+
+    /**
+     * @param ArgumentSet $old
+     * @return void
+     */
+    protected function drop(ArgumentSet &$old): void
+    {
+        foreach ($old->arguments as $name => $argument) {
+            // Recursively apply the renaming to nested inputs.
```

Comment:
Leftover from copy-paste

Reply:
Not sure I understand this one, would you mind ellaborating?

Is it the `Utils::applyEach(` section that needs removing as we no longer need to recursively loop arguments, because dropping the root argument should be sufficient?

Reply:
There is no renaming going on, the comment is clearly just copy-pasted - and no longer accurate here.

Please go through the PR and make sure every line of code actually makes sense.

Reply:
Right, understood.

Would you mind checking my latest commit, I've rearranged the code so the recursive check is only called if the "root" argument isnt an instance of DropDirective.

Reply:
Good catch, have not even thought of that extra efficiency gain.